### PR TITLE
Windows-Support: Converge on platform host checks

### PR DESCRIPTION
**Pull request by johnwparent**

### CODE REVIEW

**Comment by johnwparent** [comment]
@BetsyMcPhail PTAL when you get a moment. I may have missed a few here or there, but this pretty comprehensively transitions over to using Spack's internals to determine host platform.

File: lib/spack/llnl/util/tty/pty.py [hunk 1]
```diff
@@ -316,7 +316,6 @@ def _set_up_and_run_controller_function(
         # wait for subprocess to be running and connected.
         while not ready.value:
             time.sleep(1e-5)
-            pass
```
Comment (BetsyMcPhail) [inline lib/spack/llnl/util/tty/pty.py]: This seems unrelated to the platform checks?

File: lib/spack/spack/binary_distribution.py [hunk 2]
```diff
@@ -497,14 +497,12 @@ class NoVerifyException(spack.error.SpackError):
     """
     Raised if file fails signature verification.
     """
-    pass
```
Comment (BetsyMcPhail) [inline lib/spack/spack/binary_distribution.py]: This seems unrelated to the platform checks?

File: lib/spack/spack/binary_distribution.py [hunk 3]
```diff
@@ -497,14 +497,12 @@ class NoVerifyException(spack.error.SpackError):
...
 
 
 class NoChecksumException(spack.error.SpackError):
     """
     Raised if file fails checksum verification.
     """
-    pass
```
Comment (BetsyMcPhail) [inline lib/spack/spack/binary_distribution.py]: This seems unrelated to the platform checks?

File: lib/spack/spack/build_systems/scons.py [hunk 4]
```diff
@@ -65,7 +65,6 @@ def build_test(self):
         By default, does nothing. Override this if you want to
         add package-specific tests.
         """
-        pass
```
Comment (BetsyMcPhail) [inline lib/spack/spack/build_systems/scons.py]: This seems unrelated to the platform checks?

File: lib/spack/spack/build_systems/waf.py [hunk 5]
```diff
@@ -112,7 +112,6 @@ def build_test(self):
         By default, does nothing. Override this if you want to
         add package-specific tests.
         """
-        pass
```
Comment (BetsyMcPhail) [inline lib/spack/spack/build_systems/waf.py]: This seems unrelated to the platform checks?

File: lib/spack/spack/build_systems/waf.py [hunk 6]
```diff
@@ -122,7 +121,6 @@ def install_test(self):
...
```
Comment (BetsyMcPhail) [inline lib/spack/spack/build_systems/waf.py]: This seems unrelated to the platform checks?

File: lib/spack/spack/compiler.py [hunk 7]
```diff
@@ -588,7 +588,6 @@ def search_regexps(cls, language):
 
     def setup_custom_environment(self, pkg, env):
         """Set any environment variables necessary to use the compiler."""
-        pass
```
Comment (BetsyMcPhail) [inline lib/spack/spack/compiler.py]: This seems unrelated to the platform checks?

File: lib/spack/spack/database.py [hunk 8]
```diff
@@ -30,13 +30,6 @@
 
 import six
 
-try:
```
Comment (BetsyMcPhail) [inline lib/spack/spack/database.py]: This seems unrelated to the platform checks?

File: lib/spack/spack/package.py [hunk 9]
```diff
@@ -2129,7 +2129,6 @@ def setup_dependent_package(self, module, dependent_spec):
                 query the dependent's state.  Note that *this*
                 package's spec is available as ``self.spec``.
         """
-        pass
```
Comment (BetsyMcPhail) [inline lib/spack/spack/package.py]: This seems unrelated to the platform checks?

File: lib/spack/spack/test/cmd/build_env.py [hunk 10]
```diff
@@ -6,6 +6,7 @@
 import pytest
 from six.moves import cPickle
 
+import spack.platforms
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/build_env.py]: No longer needed

File: lib/spack/spack/test/cmd/buildcache.py [hunk 11]
```diff
@@ -68,7 +69,8 @@ def test_buildcache_list_duplicates(mock_get_specs, capsys):
     assert output.count('mpileaks') == 3
 
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Not yet implemented on windows")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
+                    reason="Install hangs on windows")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/buildcache.py]: Is the reason "Install hangs on windows" accurate or a copy + paste error? Similar for other tests in this file

File: lib/spack/spack/test/cmd/ci.py [hunk 12]
```diff
@@ -66,7 +66,8 @@ def set_env_var(key, val):
     os.environ[key] = val
 
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Error on Win")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
+                    reason="Install hangs on windows")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/ci.py]: Is the reason "Install hangs on windows" accurate or a copy + paste error? Similar for other tests in this file

File: lib/spack/spack/test/cmd/compiler.py [hunk 13]
```diff
@@ -62,7 +63,8 @@ def mock_compiler_dir(tmpdir, mock_compiler_version):
     return str(tmpdir)
 
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Error on Win")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/compiler.py:65]: Is the reason "Install hangs on windows" accurate or a copy + paste error? Similar for other tests in this file

File: lib/spack/spack/test/cmd/commands.py [hunk 14]
```diff
@@ -216,7 +216,8 @@ def test_update_completion_arg(tmpdir, monkeypatch):
     assert "--update-completion" in mock_bashfile.read()
 
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Error on Win")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/commands.py:219]: Is the reason "Install hangs on windows" accurate or a copy + paste error?

File: lib/spack/spack/test/cmd/compiler.py [hunk 15]
```diff
@@ -115,7 +117,8 @@ def test_compiler_remove(mutable_config, mock_packages):
     assert spack.spec.CompilerSpec("gcc@4.5.0") not in compilers
 
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Error on Win")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
+                    reason="Install hangs on windows")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/compiler.py]: Is the reason "Install hangs on windows" accurate or a copy + paste error? Similar for other tests in this file

File: lib/spack/spack/test/cmd/config.py [hunk 16]
```diff
@@ -240,7 +240,8 @@ def test_config_add_ordered_dict(mutable_empty_config):
 """
 
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Error on Win")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
+                    reason="Install hangs on windows")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/config.py]: Is the reason "Install hangs on windows" accurate or a copy + paste error? Similar for other tests in this file

File: lib/spack/spack/test/cmd/dev_build.py [hunk 17]
```diff
@@ -225,7 +233,8 @@ def test_dev_build_env(tmpdir, mock_packages, install_mockery,
         assert f.read() == spec.package.replacement_string
 
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Error on Win")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
+                    reason="Install hangs on windows")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/dev_build.py]: Is the reason "Install hangs on windows" accurate or a copy + paste error?

File: lib/spack/spack/test/cmd/env.py [hunk 18]
```diff
@@ -195,7 +201,8 @@ def test_env_install_single_spec(install_mockery, mock_fetch):
     assert e.specs_by_hash[e.concretized_order[0]].name == 'cmake-client'
 
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Error on Win")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
+                    reason="Install hangs on windows")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/env.py]: Is the reason "Install hangs on windows" accurate or a copy + paste error?

File: lib/spack/spack/test/cmd/graph.py [hunk 19]
```diff
@@ -3,40 +3,44 @@
...
 
+import spack.platforms
 from spack.main import SpackCommand, SpackCommandError
 
 graph = SpackCommand('graph')
 
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Error on Win")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
+                    reason="Install hangs on windows")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/graph.py]: Is the reason "Install hangs on windows" accurate or a copy + paste error? Similar for other tests in this file

File: lib/spack/spack/test/cmd/info.py [hunk 20]
```diff
@@ -37,7 +37,8 @@ def _print(*args):
     monkeypatch.setattr(spack.cmd.info.color, 'cprint', _print, raising=False)
 
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Error on Win")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
+                    reason="Install hangs on windows")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/info.py]: Is the reason "Install hangs on windows" accurate or a copy + paste error? Similar for other tests in this file

File: lib/spack/spack/test/cmd/install.py [hunk 21]
```diff
@@ -125,7 +131,8 @@ def test_install_dirty_flag(arguments, expected):
     assert args.dirty == expected
 
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Error on Win")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
+                    reason="Install hangs on windows")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/install.py]: Is the reason "Install hangs on windows" accurate or a copy + paste error? Similar for other tests in this file

File: lib/spack/spack/test/cmd/location.py [hunk 22]
```diff
@@ -49,14 +49,16 @@ def mock_spec():
     shutil.rmtree(pkg.stage.path)
 
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Error on Win")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
+                    reason="Install hangs on windows")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/location.py]: Is the reason "Install hangs on windows" accurate or a copy + paste error? Similar for other tests in this file

File: lib/spack/spack/test/cmd/mark.py [hunk 23]
```diff
@@ -16,21 +16,24 @@
 uninstall = SpackCommand('uninstall')
 
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Error on Win")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
+                    reason="Install hangs on windows")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/mark.py]: Is the reason "Install hangs on windows" accurate or a copy + paste error? Similar for other tests in this file

File: lib/spack/spack/test/cmd/mirror.py [hunk 24]
```diff
@@ -152,7 +154,8 @@ def test_exclude_file(mock_packages, tmpdir, config):
     assert (not expected_exclude & set(mirror_specs))
 
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Error on Win")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
+                    reason="Install hangs on windows")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/mirror.py]: Is the reason "Install hangs on windows" accurate or a copy + paste error? Similar for other tests in this file

File: lib/spack/spack/test/cmd/module.py [hunk 25]
```diff
@@ -60,14 +60,16 @@ def module_type(request):
 # TODO : this requires having a separate directory for test modules
 # TODO : add tests for loads and find to check the prompt format
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Error on Win")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
+                    reason="Install hangs on windows")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/module.py]: Is the reason "Install hangs on windows" accurate or a copy + paste error? Similar for other tests in this file

File: lib/spack/spack/test/cmd/pkg.py [hunk 26]
```diff
@@ -105,7 +105,8 @@ def split(output):
 pkg = spack.main.SpackCommand('pkg')
 
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Error on Win")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
+                    reason="Install hangs on windows")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/pkg.py]: Is the reason "Install hangs on windows" accurate or a copy + paste error? Similar for other tests in this file

File: lib/spack/spack/test/cmd/providers.py [hunk 27]
```diff
@@ -3,17 +3,16 @@
...
 
+import spack.platforms
 from spack.main import SpackCommand
 
 providers = SpackCommand('providers')
 
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Error on Win")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
+                    reason="Install hangs on windows")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/providers.py]: Is the reason "Install hangs on windows" accurate or a copy + paste error? Similar for other tests in this file

File: lib/spack/spack/test/cmd/uninstall.py [hunk 28]
```diff
@@ -26,23 +26,26 @@ def __init__(self, packages, all=False, force=False, dependents=False):
         self.yes_to_all = True
 
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Error on Win")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
+                    reason="Install hangs on windows")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/uninstall.py]: Is the reason "Install hangs on windows" accurate or a copy + paste error? Similar for other tests in this file

File: lib/spack/spack/test/cmd/verify.py [hunk 29]
```diff
@@ -21,7 +21,8 @@
 install = SpackCommand('install')
 
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Error on Win")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
+                    reason="Install hangs on windows")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/verify.py]: Is the reason "Install hangs on windows" accurate or a copy + paste error? Similar for other tests in this file

File: lib/spack/spack/test/cmd/versions.py [hunk 30]
```diff
@@ -3,48 +3,53 @@
...
+import spack.platforms
 from spack.main import SpackCommand
 from spack.version import Version
 
 versions = SpackCommand('versions')
 
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Error on Win")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
+                    reason="Install hangs on windows")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/versions.py]: Is the reason "Install hangs on windows" accurate or a copy + paste error? Similar for other tests in this file

File: lib/spack/spack/test/cmd/view.py [hunk 31]
```diff
@@ -26,7 +26,8 @@ def create_projection_file(tmpdir, projection):
     return projection_file
 
 
-@pytest.mark.skipif(sys.platform == 'win32', reason="Error on Win")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
+                    reason="Install hangs on windows")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/cmd/view.py]: Is the reason "Install hangs on windows" accurate or a copy + paste error? Similar for other tests in this file

File: lib/spack/spack/test/flag_handlers.py [hunk 32]
```diff
@@ -8,6 +8,7 @@
 import pytest
 
 import spack.build_environment
+import spack.platforms
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/flag_handlers.py]: No longer needed

File: lib/spack/spack/test/installer.py [hunk 33]
```diff
@@ -41,7 +41,6 @@ def _mock_repo(root, namespace):
 
 def _noop(*args, **kwargs):
     """Generic monkeypatch no-op routine."""
-    pass
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/installer.py]: Doesn't seem related to this change?

File: lib/spack/spack/test/llnl/util/filesystem.py [hunk 34]
```diff
@@ -149,7 +150,8 @@ def test_multiple_src_file_dest(self, stage):
                 fs.install('source/a/*/*', 'dest/1')
 
 
-@pytest.mark.skipif(os.name == 'nt', reason="Skip test on Windows")
+@pytest.mark.skipif(str(spack.platforms.host()) == 'windows',
+                    reason="Not yet implemented on windows")
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/llnl/util/filesystem.py]: Is the reason "Install hangs on windows" accurate or a copy + paste error? Similar for other tests in this file

File: lib/spack/spack/test/util/spack_lock_wrapper.py [hunk 35]
```diff
@@ -13,6 +13,7 @@
 from llnl.util.filesystem import group_ids
 
 import spack.config
+import spack.platforms
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/util/spack_lock_wrapper.py]: Not needed

File: lib/spack/spack/test/util/spack_yaml.py [hunk 36]
```diff
@@ -6,6 +6,7 @@
 import re
 
 import spack.config
+import spack.platforms
```
Comment (BetsyMcPhail) [inline lib/spack/spack/test/util/spack_yaml.py]: not needed

**Review by BetsyMcPhail — COMMENTED** [review]
Generally a big improvement!

It looks like there's some copy + replace errors (hopefully easily fixed with another copy + replace)

File: lib/spack/llnl/util/tty/pty.py (hunk 1 above)
Comment (johnwparent) [reply lib/spack/llnl/util/tty/pty.py]: I let spack's auto-fix run locally, this must have been the result of that, because I have no idea how else this would have been changed. The same is true for similar comments/changes. I don't think this will be a problem, but I can undo it all the same.

File: lib/spack/spack/test/cmd/buildcache.py (hunk 11 above)
Comment (johnwparent) [reply lib/spack/spack/test/cmd/buildcache.py]: I used a regex replace without thinking, I'll change all of these back.
